rest-service/manager_rest/rest/rest_utils.py

```diff
@@ -546,36 +546,42 @@ def retrieve_and_display_dependencies(self, target_id,
 
 
 def update_inter_deployment_dependencies(sm):
-    dependencies_list = sm.list(models.InterDeploymentDependencies,
-                                filters={'target_deployment': None})
+    dependencies_list = sm.list(models.InterDeploymentDependencies)
     for dependency in dependencies_list:
         if (dependency.target_deployment_func and
                 not dependency.external_target):
             _update_dependency_target_deployment(sm, dependency)
 
 
 def _update_dependency_target_deployment(sm, dependency):
-    target_deployment_id = _evaluate_target_func(dependency)
-    if target_deployment_id:
-        target_deployment_instance = sm.get(models.Deployment,
-                                            target_deployment_id)
-        dependency.target_deployment = target_deployment_instance
+    eval_target_deployment = _get_deployment_from_target_func(
+        sm, dependency.target_deployment_func, dependency.source_deployment_id)
+    if (eval_target_deployment and
+            eval_target_deployment != dependency.target_deployment):
```

Comment:
@tehasdf, is this the right way to compare between elements, or should I use their IDs?

Reply:
yeah this is fine, and with sqlalchemy i think you can even use `xx is yy`, not just equality

Reply:
Thanks! This is fine even if the element is `None`, right?

Reply:
yup